src/main/kotlin/Main.kt

```diff
@@ -1,23 +1,12 @@
 import kotlinx.cli.*
-import kotlinx.cli.ArgParser
-import kotlinx.cli.ArgType
-import kotlinx.cli.required
-import kotlinx.coroutines.runBlocking
 
-fun main(args: Array<String>) = runBlocking() {
+@OptIn(ExperimentalCli::class)
+fun main(args: Array<String>) {
     val parser = ArgParser("kotlin-cli")
-    val startHeight by parser.option(ArgType.Int, shortName = "s", description = "Start Height").required()
-
+    val fetchBlockIdCommand = FetchBlockIdCommand()
+    val fetchTxIdsCommand = FetchTxIdsCommand()
```

Comment:
You shouldn't create distinct http clients for each command, they should share the same client:

```
val apiClient = MempoolClient()
val fetchBlockIdCommand = FetchBlockIdCommand(apiClient)
val fetchTxIdsCommand = FetchTxIdsCommand(apiClient)
```